### ISSUE TITLE
Semantic based finders do not work in profile mode

### ISSUE THREAD

**Issue by dnfield** [flutter/flutter#58612]
Whether using Flutter driver, or using an e2e test, we should allow finding `bySemanticsLabel` in profile mode.

Currently, this doesn't work because the finder/commands use `debugSemantics` on `RenderObject`, which is guarded by asserts.

I propose we change this to be guarded by `kReleaseMode`.

/cc @albertwang0116 @goderbauer

**Comment by goderbauer** [comment]
> I propose we change this to be guarded by kReleaseMode.

Make it so!
